pkg/shared/families/malware/yara/yara.go

```diff
@@ -96,23 +99,38 @@ func (s *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 	return nil
 }
 
-func getRuleFilePath(compiledRuleURL string, logger *log.Entry) (string, error) {
-	parsed, err := url.Parse(compiledRuleURL)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse Yara rule URL=%s: %w", compiledRuleURL, err)
+func getCompiledRuleFilePath(cfg config.Config, logger *log.Entry) (string, error) {
+	// If both compiled rule url and raw rule sources are defined we choose the compiled one.
+	if cfg.CompiledRuleURL != "" {
+		parsed, err := url.Parse(cfg.CompiledRuleURL)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse Yara rule URL=%s: %w", cfg.CompiledRuleURL, err)
+		}
+
+		switch parsed.Scheme {
+		case "file":
+			return parsed.Path, nil
+		case "http", "https":
+			return downloadCompiledRules(cfg.CompiledRuleURL, logger)
+		default:
+			return "", errors.New("unsupported Yara rules URL")
+		}
 	}
 
-	switch parsed.Scheme {
-	case "file":
-		return parsed.Path, nil
-	case "http", "https":
-		return downloadRules(compiledRuleURL, logger)
-	default:
-		return "", errors.New("unsupported Yara rules URL")
+	if len(cfg.RuleSources) != 0 {
```

Comment:
maybe better to do it else if, to align more with the comment above?

Reply:
https://github.com/openclarity/vmclarity/pull/725/commits/9869d4ae2c9e376fac79947791e7481311b371d0